include/swift/IDE/CodeCompletionResultPrinter.h

```diff
@@ -37,8 +40,10 @@ void printCodeCompletionResultTypeNameAnnotated(
 void printCodeCompletionResultSourceText(
     const CodeCompletionResult &Result, llvm::raw_ostream &OS);
 
-void printCodeCompletionResultFilterName(
-    const CodeCompletionResult &Result, llvm::raw_ostream &OS);
+/// Print 'FilterName' from \p str into memory managed by \p Allocator and
+/// return it.
+llvm::StringRef getCodeCompletionResultFilterName(
```

Comment:
Nitpick: consistency of allocator first/last (first everywhere else, though personally I'm partial to last :P)

Reply:
I actually prefer 'lowerCamelCase' for local variable 😓

Reply:
Do you mean with `copyString`/`copyCString`? I actually prefer allocator last. I just kept the exiting interface for `copyString`. I can rewrite `copyString`/`copyCString` in this PR if you want.

Reply:
There was another one as well... I think all the `create` methods.

I also prefer last. If there's only a few places feel free, otherwise we can do it another time.

Reply:
I feel `create()` methods are different. AST's all `create*()` (e.g. `PatternBindingDecl::createImplict`) methods receives `ASTContext &` as the first argument.